core-bundle/tests/Controller/RedirectControllerTest.php

```diff
@@ -0,0 +1,125 @@
+use Contao\CoreBundle\Tests\TestCase;
+use Contao\PageModel;
+use Symfony\Bundle\FrameworkBundle\Controller\RedirectController as SymfonyRedirectController;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\ResponseHeaderBag;
+
+class RedirectControllerTest extends TestCase
+{
+    public function testAddsTheHeader(): void
+    {
+        $response = $this->createMock(RedirectResponse::class);
+
+        $response->headers = $this->createMock(ResponseHeaderBag::class);
+        $response->headers
+            ->expects($this->once())
+            ->method('set')
+            ->with('Strict-Transport-Security', 'max-age=0')
+        ;
+
+        $inner = $this->createMock(SymfonyRedirectController::class);
+        $inner
+            ->expects($this->once())
+            ->method('urlRedirectAction')
+            ->willReturn($response)
+        ;
+
+        $pageModel = $this->mockClassWithProperties(PageModel::class, ['useSSL' => false]);
+
+        $request = Request::create('https://localhost/');
+        $request->attributes->set('pageModel', $pageModel);
+
+        $controller = new RedirectController($inner);
+        $controller->urlRedirectAction($request, '/foo/bar');
+    }
+
+    public function testDoesNotAddTheHeaderForInsecureRequess(): void
+    {
+        $response = $this->createMock(RedirectResponse::class);
+
+        $response->headers = $this->createMock(ResponseHeaderBag::class);
+        $response->headers
+            ->expects($this->never())
+            ->method('set')
+        ;
+
+        $inner = $this->createMock(SymfonyRedirectController::class);
+        $inner
+            ->expects($this->once())
+            ->method('urlRedirectAction')
+            ->willReturn($response)
+        ;
+
+        $pageModel = $this->mockClassWithProperties(PageModel::class, ['useSSL' => false]);
+
+        $request = Request::create('http://localhost/');
+        $request->attributes->set('pageModel', $pageModel);
+
+        $controller = new RedirectController($inner);
+        $controller->urlRedirectAction($request, '/foo/bar');
+    }
+
+    public function testDoesNotAddTheHeaderWithoutPageModel(): void
+    {
+        $response = $this->createMock(RedirectResponse::class);
+
+        $response->headers = $this->createMock(ResponseHeaderBag::class);
+        $response->headers
+            ->expects($this->never())
+            ->method('set')
+        ;
+
+        $inner = $this->createMock(SymfonyRedirectController::class);
+        $inner
+            ->expects($this->once())
+            ->method('urlRedirectAction')
+            ->willReturn($response)
+        ;
+
+        $request = Request::create('https://localhost/');
+
+        $controller = new RedirectController($inner);
+        $controller->urlRedirectAction($request, '/foo/bar');
+    }
+
+    public function testDoesNotAddTheHeaderIfRootPageUsesSSL(): void
```

Comment:
Isn‘t that the case where we want to add the header?

Reply:
Ah, no that's actually correct. "The header" in this case refers to `Strict-Transport-Security': max-age=0' which is **unsetting** any potentially previously cached value. So if you use SSL you **do not** want the header that **unsets**. You only want the header if it should unset. 😊 
The one that actually adds it is not the `RedirectController` but teh `TransportSecurityHeaderListener`. 
So yes, it is correct.

Reply:
Ah, I see. Thanks for the explanation.